velox/row/CompactRow.cpp

```diff
@@ -17,6 +17,183 @@
 #include "velox/vector/FlatVector.h"
 
 namespace facebook::velox::row {
+namespace {
+constexpr size_t kSizeBytes = sizeof(int32_t);
+
+void writeInt32(char* buffer, int32_t n) {
+  memcpy(buffer, &n, kSizeBytes);
+}
+
+int32_t readInt32(const char* buffer) {
+  int32_t n;
+  memcpy(&n, buffer, kSizeBytes);
+  return n;
+}
+
+FOLLY_ALWAYS_INLINE void writeFixedWidth(
+    char* buffer,
+    size_t& offset,
+    const char* rawData,
+    vector_size_t index,
+    size_t valueBytes) {
+  memcpy(buffer + offset, rawData + index * valueBytes, valueBytes);
+  offset += valueBytes;
+}
+
+FOLLY_ALWAYS_INLINE void
+writeTimestamp(char* buffer, size_t& offset, const Timestamp& timestamp) {
+  // Write micros(int64_t) for timestamp value.
+  const auto micros = timestamp.toMicros();
+  memcpy(buffer + offset, &micros, sizeof(int64_t));
+  offset += sizeof(int64_t);
+}
+
+FOLLY_ALWAYS_INLINE void
+writeString(char* buffer, size_t& offset, const StringView& value) {
+  writeInt32(buffer + offset, value.size());
+  if (!value.empty()) {
+    memcpy(buffer + offset + kSizeBytes, value.data(), value.size());
+  }
+  offset += kSizeBytes + value.size();
+}
+
+// Serialize a child vector of a row type within a list of rows.
```

Comment:
"Serialize a child vector of a row type within a list of rows." can you rephrase a bit? thanks!